test/framework/src/test/java/org/elasticsearch/search/internal/ConcurrentSearchTestPluginTests.java

```diff
@@ -17,6 +18,7 @@
 
 import java.io.IOException;
 
+@LuceneTestCase.AwaitsFix(bugUrl = "todo: this is somewhat meaningless with without a worker pool")
```

Comment:
Same as above, this is verifying that we customize the number of slices while running tests, to make sure we exercise the concurrent codepath when possible.

Reply:
Right, hmmm then I can't do the short-circuit the way I did it :/ So I guess I could put in the effort to make this thing create a more "realistic" search context with a result type that gives us parallelisation here to make this work? Let me see what I can do :)

Reply:
You mean because we would have a single slice whenever an executor is not set? But don't we unset the executor only if we see a queue? I am missing something I think.

Reply:
I piggy backed on the logic for figuring out the slice count because that was looking at the `Executor` type to get the thread-count so that I had a spot that gave me the queue reference.
I also figured that it's a neat optimization to simple return a single slice when there's a queue and then go down the non-forking code path right away.
So rather than just unsetting the executor for when there's a queue, I went for unsetting the executor when there's a single slice only and setting a single slice when there's a queue. Inverting the logic a little which seemed to me optimises things a little as well? Why slice when we're not planning on parallelising in the first place? :)

Reply:
> Why slice when we're not planning on parallelising in the first place? :)

We would not slice if an executor isn't provided anyways. I will look more closely.

Reply:
I looked, your logic looks correct to me, but I don't understand why this test and the other one need to be skipped. We have a test only minimum docs per slice setting, that gets lowered to 1 in tests to ensure that we do get slices during tests and we leverage the concurrent codepath (it gets passed to the index searcher whenever an executor is also provided. That is different from the max number of slices that you now use to decide whether you provide the executor or not.  What am I missing?

Reply:
> What am I missing?

When I create the searcher in the non-concurrent path here https://github.com/elastic/elasticsearch/pull/111099/files#diff-5b3b75a4c26494593431361aefadf3faf536c865658dcbfd59a9b446ddc57cb3R190 it makes it so that the return of the max slices getter on the searcher just returns `-1`. So either we need to adjust the test here in some form or the prod code :)

Reply:
I understand a bit better,  that calling the search constructor that does not take an executor, min docs slice (that differs from max slice) is set to `-1`, which is what gets checked in this test. I guess that the problem is the more dynamic behavior of the decision making, as well as the fact that there are times where we do have an executor but with 1 max slice we don't provide it to the searcher.

I can't yet entirely follow where this test causes failures: do we end up with 1 max slice when concurrent search is enabled and we should have 1 min docs per slice?

Reply:
> I can't yet entirely follow where this test causes failures: do we end up with 1 max slice when concurrent search is enabled and we should have 1 min docs per slice?

Both paths return -1 for the min docs per slice so both fail with:

```

java.lang.AssertionError: 
Expected :50000
Actual   :-1
<Click to see difference>


	at __randomizedtesting.SeedInfo.seed([C73E4499C846CCD8:536C15AC617DAB14]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.elasticsearch.search.internal.ConcurrentSearchTestPluginTests.testConcurrentSearch(ConcurrentSearchTestPluginTests.java:37)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
```

... either expecting 50k or 1, but they always see -1 because the type in the search context is set to `NONE` and we now take the non-executor constructor path in this scenario.

Reply:
@javanna couldn't find a neat way of testing this all the way through the search context. But I think that might not matter? All we want to test is that the setting is taken up? (if you agree, that's how I made it work now :))